plugins/PrivacyManager/Model/LogDataAnonymizations.php

```diff
@@ -0,0 +1,302 @@
+                    $entry['sites'][] = 'Site ID: '. $idSite;
+                }
+            }
+        } else {
+            $entry['idsites'] = null;
+            $entry['sites'][] = 'All Websites';
+        }
+
+        if (!empty($entry['unset_visit_columns'])) {
+            $entry['unset_visit_columns'] = json_decode($entry['unset_visit_columns'], true);
+        } else {
+            $entry['unset_visit_columns'] = array();
+        }
+
+        if (!empty($entry['unset_link_visit_action_columns'])) {
+            $entry['unset_link_visit_action_columns'] = json_decode($entry['unset_link_visit_action_columns'], true);
+        } else {
+            $entry['unset_link_visit_action_columns'] = array();
+        }
+
+        $entry['anonymize_ip'] = !empty($entry['anonymize_ip']);
+        $entry['anonymize_location'] = !empty($entry['anonymize_location']);
+        $entry['idlogdata_anonymization'] = (int) $entry['idlogdata_anonymization'];
+
+        return $entry;
+    }
+
+    public function scheduleEntry($requester, $idSites, $dateString, $anonymizeIp, $anonymizeLocation, $unsetVisitColumns, $unsetLinkVisitActionColumns, $willBeStartedNow = false)
+    {
+        BaseValidator::check('date', $dateString, [new NotEmpty()]);
+
+        list($startDate, $endDate) = $this->getStartAndEndDate($dateString); // make sure valid date
+
+        if (!empty($unsetVisitColumns)) {
+            $this->logDataAnonymizer->checkAllVisitColumns($unsetVisitColumns);
+        } else {
+            $unsetVisitColumns = array();
+        }
+        if (!empty($unsetLinkVisitActionColumns)) {
+            $this->logDataAnonymizer->checkAllLinkVisitActionColumns($unsetLinkVisitActionColumns);
+        } else {
+            $unsetLinkVisitActionColumns = array();
+        }
+        if (!empty($idSites) && $idSites !== 'all') {
+            $idSites = array_map('intval', $idSites);
+            $idSites = json_encode($idSites);
+        } else {
+            $idSites = null;
+        }
+
+        if (!$anonymizeIp && !$anonymizeLocation && empty($unsetVisitColumns) && empty($unsetLinkVisitActionColumns)) {
+            throw new \Exception('Nothing is selected to be anonymized');
+        }
+
+        $db = Db::get();
+        $now = Date::now()->getDatetime();
+
+        $values = array(
+            'idsites' => $idSites,
+            'date_start' => $startDate,
+            'date_end' => $endDate,
+            'anonymize_ip' => !empty($anonymizeIp) ? 1 : 0,
+            'anonymize_location' => !empty($anonymizeLocation) ? 1 : 0,
+            'unset_visit_columns' => json_encode($unsetVisitColumns),
+            'unset_link_visit_action_columns' => json_encode($unsetLinkVisitActionColumns),
+            'scheduled_date' => Date::now()->getDatetime(),
+            'job_start_date' => $willBeStartedNow ? $now : null, // we set a start_date when executing from CLI to avoid a race condition to prevent a task from operating a scheduled entry
```

Comment:
Does this mean that if 'job_start_date' is set to `$now`, the CLI is executing the anonymization instead of a scheduled task executing it? Trying to understand the potential race condition.

Reply:
Yep exactly, it is not the best solution probably but pragmatic :)

Reply:
Gotcha, makes sense to me, the job's started by the CLI app, so it's already in progress :)